jobs/V3/basic_company_sample.rb

```diff
@@ -12,191 +12,205 @@
+		return company_created
 	rescue
 		puts "Got exception while creating company"
 	end
 	# [END create_company]
 end
 
-def get_company(company_name)
+def job_discovery_get_company company_name:
 	# [START get_company]
+	# company_name  = "The name of the company you want to get"
+
 	require "google/apis/jobs_v3"
 
 	jobs = Google::Apis::JobsV3
-	talentSolution_client = jobs::CloudTalentSolutionService.new
+	talent_solution_client = jobs::CloudTalentSolutionService.new
 	# @see https://developers.google.com/identity/protocols/application-default-credentials#callingruby
-	talentSolution_client.authorization = Google::Auth.get_application_default(
+	talent_solution_client.authorization = Google::Auth.get_application_default(
 		"https://www.googleapis.com/auth/jobs"
 	)
 
 	begin
-		company_got = talentSolution_client.get_project_company(company_name);
-		puts "Company got: " + company_got.to_json;
-		return company_got;
+		company_got = talent_solution_client.get_project_company(company_name)
+		puts "Company got: #{company_got.to_json}"
+		return company_got
 	rescue
 		puts "Got exception while getting company"
-		splitted_name = company_name.split('/');
+		splitted_name = company_name.split('/')
 		if splitted_name[0] != "projects" || splitted_name[2] != "companies" || splitted_name[1].empty? || splitted_name[3].empty?
-		puts "Invalid companyName format";
+		puts "Invalid companyName format"
 		end
 	end
 	# [END get_company]
 end
 
-def update_company(company_name, company_to_be_updated)
+def job_discovery_update_company company_name:, company_updated:
 	# [START update_company]
+	# company_name  = "The name of the company you want to update"
+	# company_updated  = "The new company object to be updated"
+
 	require "google/apis/jobs_v3"
 
 	jobs = Google::Apis::JobsV3
-	talentSolution_client = jobs::CloudTalentSolutionService.new
+	talent_solution_client = jobs::CloudTalentSolutionService.new
 	# @see https://developers.google.com/identity/protocols/application-default-credentials#callingruby
-	talentSolution_client.authorization = Google::Auth.get_application_default(
+	talent_solution_client.authorization = Google::Auth.get_application_default(
 		"https://www.googleapis.com/auth/jobs"
 	)
 
 	begin
-		update_company_request = jobs::UpdateCompanyRequest.new;
-		update_company_request.company = company_to_be_updated;
-		company_updated= talentSolution_client.patch_project_company(company_name, update_company_request);
-		puts "Company updated: " + company_updated.to_json;
-		return company_updated;
+		update_company_request = jobs::UpdateCompanyRequest.new
+		update_company_request.company = company_updated
+		company_updated= talent_solution_client.patch_project_company(company_name, update_company_request)
+		puts "Company updated: #{company_updated.to_json}"
+		return company_updated
 	rescue
 		puts "Got exception while updating company"
-		splitted_name = company_name.split('/');
+		splitted_name = company_name.split('/')
```

Comment:
Since the company name has to follow the format "projects/`project_id`/companies/`company_id`" and it's user input. The failure can be caused by bad-formatted company name. I am checking whether the user input is valid.